src/solvers/prop/prop_conv_solver.cpp

```diff
@@ -416,13 +413,11 @@ void prop_conv_solvert::add_constraints_to_prop(const exprt &expr, bool value)
         }
         else if(expr.id() == ID_implies)
         {
-          if(expr.operands().size() == 2)
-          {
-            literalt l0 = convert(expr.op0());
-            literalt l1 = convert(expr.op1());
-            prop.lcnf(!l0, l1);
-            return;
-          }
+          const auto &implies_expr = to_implies_expr(expr);
+          literalt l0 = convert(implies_expr.op0());
```

Comment:
`lhs()` and `rhs()`

Reply:
Same as above -- will do a PR for a `binary_infix_exprt`.